python/sglang/bench_serving.py

```diff
@@ -1670,9 +1670,20 @@ def calculate_metrics(
                 tokenizer.encode(outputs[i].generated_text, add_special_tokens=False)
             )
             retokenized_output_lens.append(retokenized_output_len)
-            total_input += input_requests[i].prompt_len
-            total_input_text += input_requests[i].text_prompt_len
-            total_input_vision += input_requests[i].vision_prompt_len
+            if isinstance(input_requests[i], dict):
+                # Mooncake case
+                prompt_len = (
+                    outputs[i].prompt_len
+                    if outputs[i].prompt_len > 0
+                    else input_requests[i].get("input_length", 0)
+                )
+                total_input += prompt_len
+                total_input_text += prompt_len
+                total_input_vision += 0
```

Comment:
Consider other cases? If we have more datasets, but the format is not the same as mooncake dataset.